qiita_db/sql_connection.py

```diff
@@ -504,140 +410,212 @@ def execute_fetchone(self, sql, sql_args=None):
-        -------
-        Tuple
-            The results of the fetchone query
-        dict
-            dictionary in the form of {column: type}
+        queue_name : str
+            The name of the queue
 
         Raises
         ------
-        QiitaDBExecutionError
-            if there is some error executing the SQL query
+        KeyError
+            If the queue `queue_name` does not exist in the handler
+        """
+        if queue_name not in self.queues:
+            raise KeyError("Queue '%s' does not exist" % queue_name)
 
-        Notes
-        -----
-        from psycopg2 documentation, only variable values should be bound
-        via sql_args, it shouldn't be used to set table or field names. For
-        those elements, ordinary string formatting should be used before
-        running execute.
+    def create_queue(self, queue_name):
+        """Add a new queue to the connection
+
+        Parameters
+        ----------
+        queue_name : str
+            Name of the new queue
+
+        Raises
+        ------
+        KeyError
+            Queue name already exists
         """
-        with self._sql_executor(sql, sql_args) as pgcursor:
-            result = pgcursor.fetchone()
-            types = {desc[0]: self.TYPE_CODES[desc[1]]
-                     for desc in pgcursor.description}
+        if queue_name in self.queues:
+            raise KeyError("Queue %s already exists" % queue_name)
 
-        return result, types
+        self.queues[queue_name] = []
 
-    def execute(self, sql, sql_args=None):
-        """ Executes an SQL query with no results
+    def list_queues(self):
+        """Returns list of all queue names currently in handler
+
+        Returns
+        -------
+        list of str
+            names of queues in handler
+        """
+        return self.queues.keys()
+
+    def add_to_queue(self, queue, sql, sql_args=None, many=False):
+        """Add an sql command to the end of a queue
 
         Parameters
         ----------
+        queue : str
+            name of queue adding to
         sql : str
-            The SQL query
-        sql_args : tuple or list, optional
-            The arguments for the SQL query
+            sql command to run
+        sql_args : list, tuple or dict, optional
+            the arguments to fill sql command with
+        many : bool, optional
+            Whether or not this should be treated as an executemany command.
+            Default False
 
         Raises
         ------
-        QiitaDBExecutionError
-            if there is some error executing the SQL query
+        KeyError
+            queue does not exist
 
-        Note: from psycopg2 documentation, only variable values should be bound
-            via sql_args, it shouldn't be used to set table or field names. For
-            those elements, ordinary string formatting should be used before
-            running execute.
+        Notes
+        -----
+        Queues are executed in FIFO order
```

Comment:
This note seems unnecessary, since queues are by definition FIFO, but I guess it wouldn't hurt to keep it here